src/triage/component/audition/thresholding.py

```diff
@@ -21,6 +21,66 @@ def _of_metric(df, metric_filter):
     ]
 
 
+class ModelGroupChecker(object):
+    def __init__(
+        self,
+        train_end_times,
+        initial_model_group_ids,
+        models_table,
+        db_engine
+    ):
+        """Before create the distance table, we want to check if the model_group_ids
+        and train_end_times passed in are reasonable to prevent incomparable model groups
+        from being populated to the distance table.
+
+        Args:
+            train_end_times (list) The set of train end times to consider during iteration
+            initial_model_group_ids (list) The initial list of model group ids to narrow down
+            models_table (string) The name of the results schema
+            db_engine (sqlalchemy.engine) A database engine with access to a results schema
+                of a completed modeling run
+        """
+        self.train_end_times = train_end_times
+        self._initial_model_group_ids = initial_model_group_ids
```

Comment:
Any reason for this one to be "protected" (with a leading underscore), unlike the others?